packages/aws-cdk-lib/aws-stepfunctions/lib/states/choice.ts

```diff
@@ -92,13 +92,14 @@ export class Choice extends State {
   /**
    * Return the Amazon States Language object for this state
    */
-  public toStateJson(queryLanguage?: QueryLanguage): object {
+  public toStateJson(topLevelQueryLanguage?: QueryLanguage): object {
     return {
       Type: StateType.CHOICE,
-      ...this.renderQueryLanguage(queryLanguage),
+      ...this.renderQueryLanguage(topLevelQueryLanguage),
       Comment: this.comment,
       ...this.renderInputOutput(),
       ...this.renderChoices(),
+      ...this.renderAssign(topLevelQueryLanguage),
```

Comment:
Choice state itself does not support `Assign`. The conditions support it.

> The Assign field is available at the top level of each state (except Succeed and Fail), inside Choice state rules

https://docs.aws.amazon.com/step-functions/latest/dg/workflow-variables.html#assign-field-in-asl

Reply:
This is new to me so I was wrong apologies. Choice supports `Assign` but the top level assign is only evaluated in the default case where rules match

Reply:
Nice catch! I added assign and outputs properties to condition options.